### PR TITLE
Move objectPerformance contents to new structure

### PR DESCRIPTION
**Pull request by danielhundhausen**
All the contents of `objectPerformance` were move to the appropriate places in the new structure of the python project. `caching` and `object_performance` are now split into two distinct submodules of `menu_tools`. The functionality of caching and `object_performance` was tested and is unchanged.

The from `object_performance` is now saved at `outputs/object_performance`.

The configuration files can now be found under `configs/object_performance_plots`.

### CODE REVIEW

**Comment by artlbv** [comment]
Thanks!
I think we wanted to have a common config for each version, e.g. `V29` containting `cache/objects/rates`. 
Do you want to change that later or should it be changed here?

File: pyproject.toml
Comment (artlbv) [inline pyproject.toml:1]: is this file needed?

**Comment by danielhundhausen** [comment]
> Thanks! I think we wanted to have a common config for each version, e.g. `V29` containting `cache/objects/rates`. Do you want to change that later or should it be changed here?

I have updated this.

File: pyproject.toml
Comment (danielhundhausen) [reply pyproject.toml:1]: I think this is the modern version `setup.py`. [Here](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#) is the documentation regarding it. I think we can get rid of either `setup.py` or `pyproject.toml` if we so desire. I had it in `objectPerformance` up till now and only moved it one layer up.

**Comment by danielhundhausen** [comment]
Since there is a new awkward version that fixes the bug we found, I have updated the ak version to 2.5.2 and produced [a number of comparison plots](https://dhundhausen.web.cern.ch/L1_PhaseII/comp_outputV29-ak25vsV29-reproduce-main-incl-cache/) to the old version that confirms that everything still closes.
